packages/lesmis/src/components/SearchBar/useAutocomplete.js

```diff
@@ -0,0 +1,67 @@
+/*
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ *
+ */
+import useMuiAutocomplete from '@material-ui/lab/useAutocomplete';
+
+export function createFilterOptions(config = {}) {
+  const { limit, ignoreCase = true, stringify, trim = false } = config;
+
+  return (options, { inputValue, getOptionLabel }) => {
+    if (inputValue === '') {
+      // Todo: show recent searches
+      //  @see https://app.zenhub.com/workspaces/active-sprints-5eea9d3de8519e0019186490/issues/beyondessential/tupaia-backlog/2495
+      return [];
+    }
+
+    let input = trim ? inputValue.trim() : inputValue;
+    if (ignoreCase) {
+      input = input.toLowerCase();
+    }
+
+    const fromStartMatches = options.filter(option => {
+      let candidate = (stringify || getOptionLabel)(option);
+      if (ignoreCase) {
+        candidate = candidate.toLowerCase();
+      }
+
+      return candidate.indexOf(input) === 0;
+    });
+
+    const anyMatches = options.filter(option => {
+      let candidate = (stringify || getOptionLabel)(option);
+      if (ignoreCase) {
+        candidate = candidate.toLowerCase();
+      }
+
+      return candidate.substring(1).indexOf(input) > -1;
+    });
+
+    const filteredOptions = [...fromStartMatches, ...anyMatches];
+
+    return filteredOptions.slice(0, limit);
```

Comment:
Hi Tom, sorry I started the review for this yesterday and never got to finish it 😢 

@edmofro has touched on pretty much everything else I was gonna say, just one more thing of note.

More of a nice to have I think, but could be nice to optimise this code a little bit by returning early when we reach the limit, eg something like:
```
const anyMatches = [];
const filteredOptions = [];

for (let i = 0; i < options.length && filteredOptions.length <= limit; i++) {
  const option = options[i];
  let candidate = (stringify || getOptionLabel)(option);
  if (ignoreCase) {
    candidate = candidate.toLowerCase();
  }
  
  if (candidate.indexOf(input) === 0) {
    filteredOptions.push(option); // Matches start
  } else if (candidate.substring(1).indexOf(input) > -1)
    anyMatches.push(option); // Matches anywhere
  }
}

if (filteredOptions.length === limit) {
  return filteredOptions;
}

return [...filteredOptions, ...anyMatches].slice(0, limit);
```